runtime/queries/metricsview_aggregation.go

```diff
@@ -525,6 +528,14 @@ func (q *MetricsViewAggregation) getDisplayName(ctx context.Context, rt *runtime
 	return resourceName.Name, nil
 }
 
+// timeseries request will not have any limit set, so we should not set any default limit if time range is specified otherwise timeseries graph will be partial
+func (q *MetricsViewAggregation) shouldAddLimit() bool {
```

Comment:
This is a very simple fix, should we do better limit calculation in metricsview package after time range is resolved and look for computed time dimension in the query and use its grain to calculate the limit.